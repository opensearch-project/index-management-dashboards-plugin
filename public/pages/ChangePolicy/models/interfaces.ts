/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Policy } from "../../../../models/interfaces";

export interface PolicyOption {
  label: string;
  value: Policy;
}
