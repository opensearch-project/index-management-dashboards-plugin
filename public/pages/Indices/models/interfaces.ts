/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Direction } from "@elastic/eui";
import { Policy } from "../../../../models/interfaces";
import { ManagedCatIndex } from "../../../../server/models/interfaces";

export interface PolicyOption {
  label: string;
  policy?: Policy;
}

export interface IndicesQueryParams {
  from: number;
  size: number;
  search: string;
  sortField: keyof ManagedCatIndex;
  sortDirection: Direction;
  showDataStreams: boolean;
}
