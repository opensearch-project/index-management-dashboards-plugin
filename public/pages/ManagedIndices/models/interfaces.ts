/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Direction } from "@elastic/eui";
import { ManagedIndexItem } from "../../../../models/interfaces";

export interface ManagedIndicesQueryParams {
  from: number;
  size: number;
  search: string;
  sortField: keyof ManagedIndexItem;
  sortDirection: Direction;
  showDataStreams: boolean;
}
