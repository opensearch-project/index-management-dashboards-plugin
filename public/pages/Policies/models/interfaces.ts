/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Direction } from "@elastic/eui";

export interface PolicyItem {
  id: string;
  seqNo: number;
  primaryTerm: number;
  policy: object; // only dumped to view as JSON as of now, don't need to type
}

export interface PoliciesQueryParams {
  from: number;
  size: number;
  search: string;
  sortField: keyof PolicyItem;
  sortDirection: Direction;
}
