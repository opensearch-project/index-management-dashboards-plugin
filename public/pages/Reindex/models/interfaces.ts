/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Conflicts } from "elasticsearch";

export interface ReindexRequest {
  waitForCompletion: boolean;
  slices?: number | string;
  maxDocs?: number;
  body: {
    conflicts?: Conflicts;
    source: {
      index: string;
      [key: string]: any;
    };
    dest: {
      index: string;
      pipeline?: string;
      op_type?: string;
    };
  };
  script?: {
    source: string;
    lang: string;
  };
}
export interface ReindexResponse {
  task: string;
}

export interface IndexSelectItem {
  status?: string;
  health?: string;
  isIndex?: boolean;
  isDataStream?: boolean;
  isAlias?: boolean;
  indices?: string[];
  writingIndex?: string;
}
