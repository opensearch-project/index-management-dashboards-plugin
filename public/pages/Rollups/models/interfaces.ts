/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Direction } from "@elastic/eui";
import { DocumentRollup } from "../../../../models/interfaces";

export interface RollupQueryParams {
  from: number;
  size: number;
  search: string;
  sortField: keyof DocumentRollup;
  sortDirection: Direction;
}
