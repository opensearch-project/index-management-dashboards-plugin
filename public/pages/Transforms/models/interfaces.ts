/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Direction } from "@elastic/eui";
import { DocumentTransform } from "../../../../models/interfaces";

export interface TransformQueryParams {
  from: number;
  size: number;
  search: string;
  sortField: keyof DocumentTransform;
  sortDirection: Direction;
}
