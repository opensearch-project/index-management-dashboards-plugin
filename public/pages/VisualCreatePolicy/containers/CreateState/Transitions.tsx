/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { EuiButton, EuiFormRow, EuiDragDropContext, EuiDroppable, EuiSpacer, EuiText, DropResult } from "@elastic/eui";
import EuiFormCustomLabel from "../../components/EuiFormCustomLabel";
import DraggableItem from "../../components/DraggableItem";
import { UITransition } from "../../../../../models/interfaces";
import { getConditionContent } from "../../utils/helpers";

interface TransitionsProps {
  transitions: UITransition[];
  onClickDeleteTransition: (idx: number) => void;
  onClickEditTransition: (transition: UITransition) => void;
  onDragEndTransitions: (dropResult: DropResult) => void;
  onClickAddTransition: () => void;
}

const Transitions = ({
  transitions,
  onClickDeleteTransition,
  onClickEditTransition,
  onDragEndTransitions,
  onClickAddTransition,
}: TransitionsProps) => {
  return (
    <>
      <EuiFormCustomLabel
        title="Transitions"
        helpText="Transitions define the conditions that need to be met for a state to change. After all actions in the current state are completed, the policy starts checking the conditions for transitions."
      />
      {!!transitions.length && (
        <EuiFormRow fullWidth isInvalid={false} error={null}>
          <EuiDragDropContext onDragEnd={onDragEndTransitions}>
            <EuiDroppable droppableId="STATE_TRANSITIONS_DROPPABLE_AREA">
              {transitions.map((transition, idx) => (
                <DraggableItem
                  key={transition.id}
                  id={transition.id}
                  content={getConditionContent(transition.transition)}
                  idx={idx}
                  isLast={transitions.length - 1 === idx}
                  onClickDelete={() => onClickDeleteTransition(idx)}
                  onClickEdit={() => onClickEditTransition(transition)}
                  draggableType="transition"
                />
              ))}
            </EuiDroppable>
          </EuiDragDropContext>
        </EuiFormRow>
      )}

      <EuiSpacer size="s" />

      {!transitions.length && (
        <EuiText>
          <p style={{ backgroundColor: "#F5F7FA", padding: "5px" }}>
            <span style={{ color: "grey", fontWeight: 200, fontSize: "15px" }}>No transitions have been added.</span>
          </p>
        </EuiText>
      )}

      <EuiSpacer />

      <EuiButton onClick={onClickAddTransition}>+ Add Transition</EuiButton>
    </>
  );
};

export default Transitions;
