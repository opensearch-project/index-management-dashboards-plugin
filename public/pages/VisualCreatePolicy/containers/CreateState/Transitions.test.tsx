/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import Transitions from "./Transitions";
import { UITransition } from "../../../../../models/interfaces";

describe("<Transitions /> spec", () => {
  it("renders the component", () => {
    const transitions: UITransition[] = [
      { id: "some_id_1", transition: { state_name: "some_state", conditions: { min_index_age: "30d" } } },
      { id: "some_id_2", transition: { state_name: "some_state", conditions: { min_size: "50gb" } } },
    ];
    const { container } = render(
      <Transitions
        transitions={transitions}
        onClickDeleteTransition={() => {}}
        onClickEditTransition={() => {}}
        onDragEndTransitions={() => {}}
        onClickAddTransition={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
