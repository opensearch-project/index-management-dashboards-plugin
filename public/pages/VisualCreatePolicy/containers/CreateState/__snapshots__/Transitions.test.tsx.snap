// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transitions /> spec renders the component 1`] = `
<div
  class="euiText euiText--medium"
  style="margin-bottom: 5px;"
>
  <h5
    class="euiFormLabel "
    style="margin-bottom: 2px;"
  >
    Transitions
  </h5>
  <p>
     
    <span
      style="font-weight: 200; font-size: 12px;"
    >
      Transitions define the conditions that need to be met for a state to change. After all actions in the current state are completed, the policy starts checking the conditions for transitions.
       
    </span>
  </p>
</div>
`;
